db/database.go

```diff
@@ -303,9 +302,26 @@ func NewDatabaseContext(dbName string, bucket base.Bucket, autoImport bool, opti
 
 			// TODO: invoke the same callback function from there as well, to pick up the auto-online handling
 
-		}); err != nil {
+		})
+
+		if err != nil {
+			// In the case of an error, unlock the change cache despite the fact it's in an unitialized state.
+			context.changeCache.StartupUnlock()
 			return nil, err
 		}
+
+		// Find the current global doc sequence and use that for the initial sequence for the change cache
+		lastSequence, err := context.LastSequence()
+		if err != nil {
+			return nil, err
```

Comment:
Need to unlock the changeCache on error here as well?

Reply:
Good catch. Fixed.